test/utils/getCalendarMonthWeeks_spec.js

```diff
@@ -157,4 +187,86 @@ describe('getCalendarMonthWeeks', () => {
+    const january2017End = january2017Start.clone().endOf('month'); // Tuesday
+
+    it('month starts at [0][0] when first day is Sunday and first day of week is Sunday', () => {
+      const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start, false, 0); // 0: Sun
+      const firstDayOfMonth = weeksInJanuary2017[0][0];
+      const rightPosition = january2017Start.isSame(firstDayOfMonth, 'day');
+      expect(rightPosition).to.equal(true);
+    });
+
+    it('month ends at [n][2] when last day is Tuesday and first day of week is Sunday', () => {
+      const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start, false, 0); // 0: Sun
+      const lastDayOfMonth = weeksInJanuary2017[weeksInJanuary2017.length - 1][2];
+      const rightPosition = january2017End.isSame(lastDayOfMonth, 'day');
+      expect(rightPosition).to.equal(true);
+    });
+
+    it('month starts at [0][4] when first day is Sunday and first day of week is Wednesday', () => {
+      const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start, false, 3); // 3: Wed
+      const firstDayOfMonth = weeksInJanuary2017[0][4];
+      const rightPosition = january2017Start.isSame(firstDayOfMonth, 'day');
+      expect(rightPosition).to.equal(true);
+    });
+
+    it('month ends at [n][6] when last day is Tuesday and first day of week is Wednesday', () => {
+      const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start, false, 3);// 3: Wed
+      const lastDayOfMonth = weeksInJanuary2017[weeksInJanuary2017.length - 1][6];
+      const rightPosition = january2017End.isSame(lastDayOfMonth, 'day');
+      expect(rightPosition).to.equal(true);
+    });
+  });
+
+  describe('firstDayOfWeek default value locale-aware', () => {
+    // using these known dates to see if they appear at the right position of the calendar
+    // trying different firstDayOfWeek values
+    const january2017Start = moment('2017-01-01'); // Sunday
+    const january2017End = january2017Start.clone().endOf('month'); // Tuesday
+
+    describe('locale with Sunday as first day of week', () => {
+      before(() => {
+        moment.locale('en');
+      });
+
+      it('month starts at [0][0] if first day is Sunday', () => {
+        const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start);
+        const firstDayOfMonth = weeksInJanuary2017[0][0];
+        const rightPosition = january2017Start.isSame(firstDayOfMonth, 'day');
+        expect(rightPosition).to.equal(true);
+      });
+
+      it('month ends at [n][2] if last day is Tuesday', () => {
+        const weeksInJanuary2017 = getCalendarMonthWeeks(january2017Start);
+        const lastDayOfMonth = weeksInJanuary2017[weeksInJanuary2017.length - 1][2];
+        const rightPosition = january2017End.isSame(lastDayOfMonth, 'day');
+        expect(rightPosition).to.equal(true);
+      });
+    });
+
+    describe('locale with Monday as first day of week', () => {
+      before(() => {
```

Comment:
we need to use `beforeEach` here; `before` is beforeAll, and we don't want that.